rct229/rules/section5/section5rule45.py

```diff
@@ -0,0 +1,75 @@
+from rct229.rule_engine.rule_base import RuleDefinitionListIndexedBase, RuleDefinitionBase
+from rct229.rule_engine.user_baseline_proposed_vals import UserBaselineProposedVals
+from rct229.ruleset_functions.compare_schedules import (compare_schedules, LEAP_YEAR_HOURS, REGULAR_YEAR_HOURS)
+
+
+class Section5Rule45(RuleDefinitionListIndexedBase):
+    """Rule 45 of ASHRAE 90.1-2019 Appendix G Section 5 (Envelope)"""
+
+    def __init__(self):
+        super(Section5Rule45, self).__init__(
+            rmrs_used=UserBaselineProposedVals(False, True, True),
+            each_rule=Section5Rule45.ZoneRule(),
+            index_rmr="baseline",
+            id="5-45",
+            description="The infiltration schedules are the same in the proposed RMR as in the baseline RMR.",
+            list_path="ruleset_model_instances[0].buildings[*].building_segments[*].zones[*]",
+            required_fields={
+                "$": ["calendar"],
+                "calendar": ["is_leap_year"]
+            }
+        )
+
+    def create_data(self, context, data=None):
+        rmr_proposed = context.proposed
+        return {"is_leap_year": rmr_proposed["calendar"]["is_leap_year"]}
+
+    def list_filter(self, context_item, data=None):
+        # baseline should match to proposed
+        zone_p = context_item.proposed
+        return (
+                zone_p.get("infiltration") is not None
+        )
+
+    class ZoneRule(RuleDefinitionBase):
+        def __init__(self):
+            super(Section5Rule45.ZoneRule, self).__init__(
+                rmrs_used=UserBaselineProposedVals(False, True, True),
+                required_fields={
+                    "infiltration": ["multiplier_schedule"]
+                },
+            )
+
+        def get_calc_vals(self, context, data=None):
+            zone_b = context.baseline
+            zone_p = context.proposed
+
+            compare_schedules_result_dictionary = {
+                "total_hours_compared": 0.0,
+                "total_hours_match": 0.0,
+                "eflh_difference": 0.0
+            }
+            mask_schedule = [1.0] * (LEAP_YEAR_HOURS if data["is_leap_year"] else REGULAR_YEAR_HOURS)
+
+            infiltration_p = zone_p["infiltration"]
+            if zone_b:
```

Comment:
Two issues here.
1. We are indexing on baseline, so `zone_p` is the "matched" one of the two.
2. Issue one doesn't matter because `zone_b` and `zone_p` will not be None. Any non-matching proposed zones are flagged in the output and `get_calc_vals` will not be called for such zones.